File: algo-c-to-_/examples/dragoncurve_.py
#
#	from src/dragon2.c
#
#	a part of main				to	dragonCurve
#

from svgplot import svgPlot
from dragoncurve import dragonCurve

def sampleWriter(pathPrefix, x, y):
    plotter = svgPlot(x, y)

    def sample(fh, order, x0, y0):
        plotter.plotStart(fh)
        dragonCurve(plotter, order, x0, y0)
        plotter.plotEnd()

    def writer(order, x0, y0):
        with open("{0}{1}.svg".format(pathPrefix, order), "w") as fh:
            sample(fh, order, x0, y0)

    return writer

#

x, y, x0, y0 = 510, 350, 120, 120
writer = sampleWriter("results/dragoncurve-py", x, y)
for order in range(1, 11):
    writer(order, x0, y0)

File: porting-from-algo-c/examples/dragoncurve.lua
--
--  from src/dragon.c
--
--    void dragon(int, double, double, int)  to  dragonCurveR
--

local M = require 'svgplot'

local dragonCurve = require 'dragoncurve'.dragonCurve
local file = require '_helper'.file

function gBody(n, dx, dy, sign, x0, y0)
	return function (plotter)
		plotter:pathStart()
		dragonCurve(plotter, n, dx, dy, sign, x0, y0)
		plotter:pathEnd(false, M.SV.PRESET_PLAIN)
	end
end

do
	local x, y, dx, dy = 400, 250, 200, 0

	local plt = M.SvgPlot(x, y)

	local pathPrefix = "results/dragoncurve"

	file(("%s-A.svg"):format(pathPrefix), "w", function (fh)
		plt:write(fh, gBody(10, dx, dy, 1, 100, 100))
	end)

	file(("%s-B.svg"):format(pathPrefix), "w", function (fh)
		plt:write(fh, gBody(12, dx, dy, -1, 70, 160))
	end)
end
